src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -339,29 +352,20 @@ static async ValueTask<FlushResult> WriteDataAndTrailersAsyncCore(Http2FrameWrit
         |                           Padding (*)                       ...
         +---------------------------------------------------------------+
     */
-    private void WriteDataUnsynchronized(int streamId, in ReadOnlySequence<byte> data, long dataLength, bool endStream)
+    private void WriteDataUnsynchronized(Http2Stream stream, in ReadOnlySequence<byte> data, long dataLength, bool endStream)
     {
```

Comment:
Nit: we could `DecrementActiveClientStreamCount` in the `endStream` case since we already know we have enough flow control availability instead of waiting until copying the last frame in `WriteDataUnsynchronizedCore`.